ehsm_kms_service/secret_manager_apis.js

```diff
@@ -234,9 +234,68 @@ const createSecret = async (res, appid, payload, DB) => {
     }
 }
 
+//update the description of secret
+const updateSecretDesc = async (res, appid, payload, DB) => {
+    //get and check param in payload
+    let secretName = getParam_String(payload, 'secretName')
+    let description = getParam_String(payload, 'description')
+    if (!checkStringParam(secretName, true)) {
+        res.send(_result(400, 'secretName cannot be empty and must be string'))
+        return
+    }
+    if (!checkStringParam(description, true)) {
+        res.send(_result(400, 'description cannot be empty and must be string'))
+        return
+    }
+
+    //Query the description through secret name and update the description
+    const secret_name_query = {
+        selector: {
+            appid,
+            secretName
+        },
+        fields: [
+            '_id',
+            '_rev',
+            'appid',
+            'secretName',
+            'encryptionKeyId',
+            'description',
+            'createTime',
+            'deleteTime',
+            'plannedDeleteTime',
+            'rotationInterval',
+            'lastRotationDate',
+            'nextRotationDate'
+        ],
+        limit: 1,
+    }
+    await DB.partitionedFind('secret_metadata', secret_name_query)
+        .then(async (secret_metadata_res) => {
```

Comment:
The async used here, makes this update multi-thread?
How can we keep the data‘s consistency while using multi-thread? Do we have any lock if a put and a update comes in at the same time?

Reply:
async has no effect here and will be deleted

Reply:
OK, the database lock issue can be fixed after we discuss it, maybe in other patches.